code/mob/living/critter/ai/spider.dm

```diff
@@ -21,11 +21,42 @@
 	transition_tasks += holder.get_instance(/datum/aiTask/timed/wander/critter/aggressive, list(holder, src))
 	transition_tasks += holder.get_instance(/datum/aiTask/sequence/goalbased/critter/attack, list(holder, src))
 	transition_tasks += holder.get_instance(/datum/aiTask/sequence/goalbased/critter/scavenge, list(holder, src))
-	transition_tasks += holder.get_instance(/datum/aiTask/succeedable/critter/vomit_egg, list(holder, src))
+	transition_tasks += holder.get_instance(/datum/aiTask/sequence/goalbased/critter/vomit_egg, list(holder, src))
 
-/datum/aiTask/succeedable/critter/vomit_egg //ew clownspiders are gross
+/datum/aiTask/sequence/goalbased/critter/vomit_egg
 	name = "clown spider vomit egg"
+	weight = 1
+	max_dist = 5
+	can_be_adjacent_to_target = FALSE
+	var/max_spiders = 4
+
+/datum/aiTask/sequence/goalbased/critter/vomit_egg/New(parentHolder, transTask)
+	..(parentHolder, transTask)
+	add_task(holder.get_instance(/datum/aiTask/succeedable/critter/vomit_egg, list(holder)))
+
+/datum/aiTask/sequence/goalbased/critter/vomit_egg/precondition()
+	var/mob/living/critter/spider/clownqueen/C = holder.owner
+	var/datum/targetable/critter/vomitegg/egg = C.abilityHolder.getAbility(/datum/targetable/critter/vomitegg)
+	if(!egg)
+		egg = C.abilityHolder.getAbility(/datum/targetable/critter/vomitegg/cluwne)
+	return !egg.disabled && (length(C.babies) < max_spiders)
```

Comment:
Clownspiders don't remove themselves from this list on death, which is a GC issue and will mean this makes them only ever spawn 4 total.

Reply:
yeah I found this out while I was testing it but I dunno how to fix